local.c

```diff
@@ -1480,6 +1480,21 @@ static unsigned int is_global_attr(struct iio_channel *chn, const char *attr)
 
 	len = ptr - attr;
 
+	// Check for matching global differential attr, like "voltage-voltage"
+	dashptr = strchr(attr, '-');
+	if (dashptr && dashptr > attr && dashptr < ptr) {
+		unsigned int len1 = dashptr - attr,
+					 len2 = ptr - dashptr - 1;
```

Comment:
Things could look more aligned if you declare len2 along with the type:
`unsigned int len2 = ptr - dashptr - 1;`